libtransact/src/state/merkle/sql/operations/list_leaves.rs

```diff
@@ -34,597 +33,99 @@ pub trait MerkleRadixListLeavesOperation {
-                    .filter(
-                        merkle_radix_state_root_leaf_index::from_state_root_id
-                            .le(state_root_id)
-                            .and(merkle_radix_state_root_leaf_index::tree_id.eq(tree_id))
-                            .and(
-                                merkle_radix_state_root_leaf_index::to_state_root_id
-                                    .is_null()
-                                    .or(merkle_radix_state_root_leaf_index::to_state_root_id
-                                        .gt(state_root_id)),
-                            ),
-                    )
-                    .select((
-                        merkle_radix_leaf::id,
-                        merkle_radix_leaf::address,
-                        merkle_radix_leaf::data,
-                        merkle_radix_state_root_leaf_index::to_state_root_id,
-                    ))
-                    .into_boxed();
-
-                if let Some(prefix) = prefix {
-                    query = query.filter(merkle_radix_leaf::address.like(format!("{}%", prefix)));
-                }
-
-                query = query
-                    .order(merkle_radix_leaf::address.asc())
-                    .then_order_by(merkle_radix_state_root_leaf_index::to_state_root_id.desc());
-
-                let leaves = query
-                    .get_results::<(i64, String, Vec<u8>, Option<i64>)>(self.conn)?
-                    .into_iter()
-                    .map(|(_, addr, data, _)| (addr, data))
-                    .collect();
-
-                Ok(leaves)
-            })
-            .map_err(|e| InternalError::from_source(Box::new(e)))
+        let results = sql_query(
+            r#"
+            WITH RECURSIVE tree_path AS
+            (
+                -- This is the initial node
+                SELECT hash, tree_id, leaf_id, children, 0 as depth
+                FROM merkle_radix_tree_node
+                WHERE hash = ? AND tree_id = ?
+
+                UNION ALL
+
+                -- Recurse through the tree
+                SELECT c.hash, c.tree_id, c.leaf_id, c.children, p.depth + 1
+                FROM merkle_radix_tree_node c, tree_path p, json_each(p.children)
+                WHERE c.hash = json_each.value
+            )
+            SELECT l.address, l.data
+            FROM tree_path t, merkle_radix_leaf l
+            WHERE t.tree_id = ? AND t.leaf_id = l.id AND l.address LIKE ?
+            ORDER BY l.address
+            "#,
+        )
+        .bind::<VarChar, _>(state_root_hash)
+        .bind::<BigInt, _>(tree_id)
+        .bind::<BigInt, _>(tree_id)
+        .bind::<VarChar, _>(format!("{}%", prefix.unwrap_or("")))
+        .load::<Leaf>(self.conn)
+        .map_err(|err| InternalError::from_source(Box::new(err)))?
+        .into_iter()
+        .map(|leaf| (leaf.address, leaf.data))
+        .collect::<Vec<_>>();
+
+        Ok(results)
     }
 }
 
-#[cfg(test)]
-mod test {
```

Comment:
Are there any simple tests that can be added here? or are they all covered in the integration tests?

Reply:
They are all covered by the integration tests